commit e6c9668f3202215ddb71617b41c19b6f05acf008
Author: Tim Kientzle <kientzle@acm.org>
Date:   Fri Jan 30 23:57:03 2015 -0800

    Add a check to archive_read_filter_consume to reject any
    attempts to move the file pointer by a negative amount.
    
    Note:  Either this or commit 3865cf2 provides a fix for
    Issue 394.

diff --git a/libarchive/archive_read.c b/libarchive/archive_read.c
index 8f71a8b..d649e9a 100644
--- libarchive/archive_read.c
+++ libarchive/archive_read.c
@@ -1471,6 +1471,8 @@ __archive_read_filter_consume(struct archive_read_filter * filter,
 {
 	int64_t skipped;
 
+	if (request < 0)
+		return ARCHIVE_FATAL;
 	if (request == 0)
 		return 0;
 
